src/NBitcoin/Block.cs

```diff
@@ -162,21 +167,22 @@ public bool CheckProofOfWork()
 
         public bool CheckProofOfWork(Consensus consensus)
         {
-            consensus = consensus ?? Network.Main.Consensus;
+            NetworkOptions options = consensus?.NetworkOptions ?? NetworkOptions.TemporaryOptions;
+            consensus = consensus ?? Network.Main.Consensus; // TODO: Not always set correctly from BlockValidator.CheckBlock..
             BigInteger bits = this.Bits.ToBigInteger();
             if ((bits.CompareTo(BigInteger.Zero) <= 0) || (bits.CompareTo(Pow256) >= 0))
                 return false;
 
             // Check proof of work matches claimed amount.
-            if (Block.BlockSignature) // Note this can only be called on a POW block.
+            if (options.IsProofOfStake) // Note this can only be called on a POW block.
                 return this.GetPoWHash() <= this.Bits.ToUInt256();
 
-            return consensus.GetPoWHash(this) <= this.Bits.ToUInt256();
+            return consensus.GetPoWHash(consensus.NetworkOptions, this) <= this.Bits.ToUInt256();
```

Comment:
We could play a bi ton this method by setting the consensus.GetPoWHash for stratis to be the x13hash
Then we can remove the `options.IsProofOfStake` check.
What do you think?

Reply:
Good idea. Made the change.